cuqi/distribution/_normal.py

```diff
@@ -42,10 +62,15 @@ def _gradient(self, val, *args, **kwargs):
             raise NotImplementedError("Gradient not implemented for distribution {} with geometry {}".format(self,self.geometry))
         if not callable(self.mean):
             return -(val-self.mean)/(self.std**2)
+        elif hasattr(self.mean, "gradient"): # for likelihood
+            model = self.mean
+            dev = val - model.forward(*args, **kwargs)
+            if isinstance(dev, numbers.Number):
+                dev = np.array([dev])
+            return model.gradient(1.0/(np.array(self.std)) @ dev, *args, **kwargs)
```

Comment:
I wonder if  "np.array" is still needed now that you have the std setter?

Reply:
Yes, `dev` is already of `np.array` type. So I removed the line 68 and 69 as suggested.